pcsx2/ps2/BiosTools.h

```diff
@@ -28,19 +31,35 @@ namespace Exception
 	};
 }
 
+#ifdef USE_HARDCODED_BIOS_INFORMATION
 struct BiosDebugInformation
 {
 	u32 biosVersion;
 	u32 biosChecksum;
 	u32 threadListAddr;
 };
 
+const extern BiosDebugInformation* CurrentBiosInformation;
+#else
+// Pattern matching for today :>
+const u32 ThreadListInstructions[3] =
+{
+	0xac420000, // sw v0,0x0(v0)
+	0x00000000, // no-op
+	0x00000000, // no-op
+};
```

Comment:
Uhhhh, is this safe? This seems like an awfully generic suite of instructions to me

Reply:
Kernel isn't too big, from my tests there is only one instance of this instruction in the bios. It gets difficult choosing instructions because they may change depending on the bios.

Reply:
works for me then!